IPython/core/completer.py

```diff
@@ -1,50 +1,13 @@
 # encoding: utf-8
 """Word completion for IPython.
 
-This module is a fork of the rlcompleter module in the Python standard
+This module started as fork of the rlcompleter module in the Python standard
 library.  The original enhancements made to rlcompleter have been sent
-upstream and were accepted as of Python 2.3, but we need a lot more
-functionality specific to IPython, so this module will continue to live as an
-IPython-specific utility.
+upstream and were accepted as of Python 2.3,
 
-Original rlcompleter documentation:
 
-This requires the latest extension to the readline module (the
-completes keywords, built-ins and globals in __main__; when completing
-NAME.NAME..., it evaluates (!) the expression up to the last dot and
-completes its attributes.
-
-It's very cool to do "import string" type "string.", hit the
-completion key (twice), and see the list of names defined by the
-string module!
-
-Tip: to use the tab key as the completion key, call
-
-    readline.parse_and_bind("tab: complete")
-
-Notes:
-
-- Exceptions raised by the completer function are *ignored* (and
-  generally cause the completion to fail).  This is a feature -- since
-  readline sets the tty device in raw (or cbreak) mode, printing a
-  traceback wouldn't work well without some complicated hoopla to save,
-  reset and restore the tty state.
-
-- The evaluation of the NAME.NAME... form may cause arbitrary
-  application defined code to be executed if an object with a
-  ``__getattr__`` hook is found.  Since it is the responsibility of the
-  application (or the user) to enable this feature, I consider this an
-  acceptable risk.  More complicated expressions (e.g. function calls or
-  indexing operations) are *not* evaluated.
-
-- GNU readline is also used by the built-in functions input() and
-  raw_input(), and thus these also benefit/suffer from the completer
-  features.  Clearly an interactive application can benefit by
-  specifying its own completer function and using raw_input() for all
-  its input.
-
-- When the original stdin is not a tty device, GNU readline is never
-  used, and this module (and the readline module) are silently inactive.
+This is now mostly Completer implementation made to function with
+prompt_toolkit, but that should still work with readline.
```

Comment:
Ok, so there are more code we can strip out. I'll remove the sentence and strip out more code later.